### ISSUE TITLE
headComment and footComment not working correctly for multiple documents

### ISSUE THREAD

**Issue by MisterMX** [mikefarah/yq#1007]
**Describe the bug**

The `headComment` operator does not work correctly if I run it for multiple documents in a file. The same goes for `footComment`.

Version of yq: 4.14.2
Operating system: linux
Installed via: binary

**Input Yaml**
```yaml
---
# head1
ab: cd
# foot1
---
# head2
ab: cd
# foot2

```

**Command**
```bash
yq e '. | headComment' test.yaml
```

**Actual behavior**
```yaml
head1
---

```

**Expected behavior**
```yaml
head1
---
head2
```

The output is the same for `footComment`:

**Command**
```bash
yq e '. | footComment' test.yaml
```

**Actual behavior**
```yaml
foot1
---

```

**Expected behavior**
```yaml
foot1
---
foot2
```

**Comment by mikefarah** [comment]
Doh I should have picked that up - sorry, will fix.

**Comment by mikefarah** [comment]
On closer inspection, this is an issue with the underling go-yaml parser (https://github.com/go-yaml/yaml/issues/610, https://github.com/go-yaml/yaml/issues/695).

"head2" is somehow being associated as the head comment of the key 'ab' and not the document node, and there's a similar issue for "foot2".

Note sure what `yq` could do about this without stuffing up other scenarios :/

**Comment by mikefarah** [comment]
As a workaround/hack, you can recursively match the nodes that have the comments attached to them:

```
yq e '... | headComment | select(. != "")' examples/data1.yaml
```

```
head1
---
head2
```

**Comment by MisterMX** [comment]
@mikefarah yes, that worked. Thanks a lot for hint 👍 

Since the bug is still prevailing I would leave this issue open until it is fixed in [go-yaml/yaml](https://github.com/go-yaml/yaml). Do you think we should open a separate bug report?

**Comment by mikefarah** [comment]
Yeah, though there hasn't been much action on that repo for a while :/
________________________________

**Comment by MisterMX** [comment]
I filed an issue. Let's see what happens: https://github.com/go-yaml/yaml/issues/801

**Comment by rriemann** [comment]
I hit the same issue when I wanted to split documents into files according to the comment content:

~~~
yq -s 'headComment | match("[^\/]+\.yaml$") | .string' export.yaml
~~~

I now use:

~~~
yq -s '... | headComment | match("[^\/]+\.yaml$") | .string' export.yaml
~~~
